configs/envoy_double_proxy.template.yaml

```diff
@@ -0,0 +1,110 @@
+---
+{% macro listener(address,ssl,proxy_proto) -%}
+- address: "{{ address }}"
+  {% if ssl -%}
+  ssl_context:
+    alpn_protocols: h2,http/1.1
+    alt_alpn_protocols: http/1.1
+    cert_chain_file: certs/servercert.pem
+    private_key_file: certs/serverkey.pem
+  {% endif -%}
+  {% if proxy_proto -%}
+  use_proxy_proto: true
+  {% endif -%}
+  filters:
+  - name: http_connection_manager
+    config:
+      codec_type: auto
+      tracing:
+        operation_name: ingress
+      idle_timeout_s: 840
+      access_log:
+      - path: "/var/log/envoy/access_error.log"
+        filter:
+          type: logical_or
+          filters:
+          - type: status_code
+            op: ">="
+            value: 500
+          - type: duration
+            op: ">="
+            value: 1000
+          - type: traceable_request
+      - path: "/var/log/envoy/access.log"
+      stat_prefix: router
+      {% if proxy_proto -%}
+      use_remote_address: true
+      {% endif -%}
+      route_config:
+        virtual_hosts:
+        - name: all
+          domains:
+          - "*"
+          routes:
+          - prefix: "/"
+            cluster: backhaul
+            timeout_ms: 20000
+      filters:
+      - name: health_check
+        config:
+          pass_through_mode: false
+          endpoint: "/healthcheck"
+      - name: buffer
+        config:
+          max_request_bytes: 5242880
+          max_request_time_s: 120
+      - name: router
+        config: {}
+{%- endmacro -%}
+listeners:
+{{ listener("tcp://0.0.0.0:9300",True,True) }}
+{{ listener("tcp://0.0.0.0:9301",False,True) }}
+admin:
+  access_log_path: "/var/log/envoy/admin_access.log"
+  address: tcp://127.0.0.1:9901
+flags_path: "/etc/envoy/flags"
+statsd_tcp_cluster_name: statsd
+tracing:
+  http:
+    driver:
+      type: lightstep
+      config:
+        access_token_file: "/etc/envoy/lightstep_access_token"
+        collector_cluster: lightstep_saas
+runtime:
+  symlink_root: "/srv/runtime_data/current"
+  subdirectory: envoy
+  override_subdirectory: envoy_override
+cluster_manager:
+  clusters:
+  - name: statsd
+    connect_timeout_ms: 250
+    type: static
+    lb_type: round_robin
+    hosts:
+    - url: tcp://127.0.0.1:8125
+  - name: backhaul
+    connect_timeout_ms: 1000
+    type: strict_dns
+    lb_type: round_robin
+    features: http2
+    max_requests_per_connection: 25000
+    ssl_context:
```

Comment:
this field is quite different now follow: https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/auth/cert.proto#envoy-api-msg-auth-upstreamtlscontext